odl/space/fspace.py

```diff
@@ -721,9 +724,9 @@ def ipow_posint(x, n):
         def power_call_out_of_place(x):
             """The power out-of-place evaluation function."""
             if p == int(p) and p >= 1:
-                return pow_posint(x_call_oop(x), int(p))
+                return pow_posint(x_call_oop(x), int(p)).astype(self.out_dtype)
```

Comment:
Hm, I have a hard time imagining a situation where this would happen, but to be sure, one could use `np.asarray(..., dtype=self.out_dtype)` instead.